src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -104,7 +106,16 @@ export default class DefaultMessagingSession implements MessagingSession {
   }
 
   private async startConnecting(reconnecting: boolean): Promise<void> {
-    const signedUrl = await this.prepareWebSocketUrl();
+    // reconnect needs to re-resolve endpoint url, which will also refresh credentials on client if they are expired
+    let endpointUrl = !reconnecting ? this.configuration.endpointUrl : null;
```

Comment:
Should we use `undefined` here as well? `this.configuration.endpointUrl` can either be string | undefined, so if that value is undefined, here endpointUrl will be undefined and not strict equal null. I think you just updated the type to use undefined on the constructor but not elsewhere.

Reply:
updated thx